lib/oli/delivery/attempts/activity_lifecycle.ex

```diff
@@ -332,21 +333,31 @@ defmodule Oli.Delivery.Attempts.ActivityLifecycle do
         }
       end)
 
-    part_input_values = Enum.join(part_input_values, ",")
-
-    sql = """
-      UPDATE part_attempts
-      SET
-        response = batch_values.response,
-        updated_at = NOW()
-      FROM (
-          VALUES
-          #{part_input_values}
-      ) AS batch_values (attempt_guid, response)
-      WHERE part_attempts.attempt_guid = batch_values.attempt_guid
-    """
-
-    Ecto.Adapters.SQL.query(Oli.Repo, sql, params)
+    part_attempt = get_part_attempt_by(attempt_guid: hd(params))
```

Comment:
The `save_part`  endpoint is a high throughput endpoint, and one that has caused performance problems in the past.  The proposed change here would add another DB call (the read to get the part attempt).  We need a solution that does not add any more DB reads or writes.  Perhaps adding into the `UPDATE` query a constraint `and lifecycle_state = :active` does the trick, handling the case of updated zero records as ""These changes could not be saved as this attempt may have already been submitted"